packages/flutter/lib/src/animation/tween.dart

```diff
@@ -256,6 +256,35 @@ class Tween<T extends dynamic> extends Animatable<T> {
   T lerp(double t) {
     assert(begin != null);
     assert(end != null);
+    assert(() {
+      // Assertions that attempt to catch common cases of tweening types
+      // that do not conform to the Tween requirements.
+      dynamic? result;
+      try {
+        result = begin + (end - begin) * t;
+        result as T;
+        return true;
+      } on NoSuchMethodError {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary('Cannot tween between $begin and $end.'),
+          ErrorDescription(
+            'The type ${begin.runtimeType} does not fully implement `+`, `-`, and/or `*`.',
+          ),
+          if (begin is Color || end is Color)
+            ErrorHint('To tween colors, use ColorTween instead.')
```

Comment:
There's a ton of possibilities here. Maybe we should list some of the others (to the extent they're visible) -- see the Tween API docs for a list that we know of. And then maybe as a final "else" we could have `There may be a dedicated "${T}Tween" class for these types, or you may need to create one.`, maybe with a link to something in the API docs that talks about how to write one (hopefully we have a section in the `Tween` docs on that, or we can add one).

Reply:
I might be a bit out of practice here, do we have a standard way to link to API docs from flutter errors?

Reply:
what you did looks great